ConfCore/Session.swift

```diff
@@ -162,4 +162,140 @@ public class Session: Object {
         }
     }
 
+    // MARK: - Decodable
+
+    private enum AssetCodingKeys: String, CodingKey {
+        case id
+        case year
+        case title
+        case downloadHD
+        case downloadSD
+        case slides
+        case hls
+        case images
+        case shelf
+        case duration
+    }
+
+    private enum SessionCodingKeys: String, CodingKey {
+        case id
+        case year
+        case title
+        case platforms
+        case description
+        case startTime
+        case eventContentId
+        case eventId
+        case media
+        case webPermalink
+        case staticContentId
+        case related
+        case track = "trackId"
+    }
+
+    private enum RelatedCodingKeys: String, CodingKey {
+        case activities
+        case resources
+    }
+
+    public convenience required init(from decoder: Decoder) throws {
+        let sessionContainer = try decoder.container(keyedBy: SessionCodingKeys.self)
+
+        let id = try sessionContainer.decode(String.self, forKey: .id)
+        let eventIdentifier = try sessionContainer.decode(String.self, forKey: .eventId)
+        let eventYear = eventIdentifier.replacingOccurrences(of: "wwdc", with: "")
+        let title = try sessionContainer.decode(String.self, forKey: .title)
+        let summary = try sessionContainer.decode(String.self, forKey: .description)
+        let trackIdentifier = try sessionContainer.decode(Int.self, forKey: .track)
+        let eventContentId = try sessionContainer.decode(Int.self, forKey: .eventContentId)
+
+        self.init()
+
+        var mediaDuration: Double?
+
+        if let assetContainer = try? sessionContainer.nestedContainer(keyedBy: AssetCodingKeys.self, forKey: .media) {
```

Comment:
Why `try?`

I'm asking about all of them because I don't want code that is so safe that it covers up bugs. Of course I understand if we're expecting failures and thus the `try?` is appropriate. Alternatively, you could do a do { } and then catch/ignore the "expected" errors and rethrow unexpected errors.

Reply:
As an example, here is my implementation of Codable for RealmOptional
```
extension RealmOptional: Encodable where Value: Encodable {

    public func encode(to encoder: Encoder) throws {

        var container = encoder.singleValueContainer()

        try value.flatMap { try container.encode($0) } ?? container.encodeNil()
    }
}

extension RealmOptional: Decodable where Value: Decodable {

    public convenience init(from decoder: Decoder) throws {
        self.init()

        let container = try decoder.singleValueContainer()

        do {
            value = try container.decode(Optional<Value>.self)
        } catch let error as DecodingError {
            switch error {
            case .valueNotFound, .keyNotFound:
                return
            default:
                throw error
            }
        }
    }
}
```